exercises/connect/.meta/src/reference/java/Connect.java

```diff
@@ -0,0 +1,75 @@
+class Connect {
+    String findWinner(String[] board) {
+        // parse
+        String[][] values = new String[board.length][board.length];
+        for (int i = 0; i < board.length; i++) {
+            values[i] = board[i].substring(i).split(" ");
+        }
+
+        int lineCount = values.length,
+            columnCount = values[0].length;
+
+        // mark starting points
+        // O starts at the top
+        for (int i = 0; i < columnCount; i++) {
+            if (values[0][i].equals("O")) values[0][i] = "o";
+        }
+        // X starts on the left
+        for (int i = 0; i < lineCount; i++) {
+            if (values[i][0].equals("X")) values[i][0] = "x";
+        }
+
+
+        // go through and connect
+        boolean changes;
+        do {
+            changes = false;
+            for (int line = 0; line < lineCount && !changes; line++) {
+                for (int column = 0; column < columnCount && !changes; column++) {
+                    if (values[line][column].equals("O") && checkSurroundingsFor(values, line, column, "o")) {
+                        values[line][column] = "o";
+                        changes = true;
+                    } else if (values[line][column].equals("X") && checkSurroundingsFor(values, line, column, "x")) {
+                        values[line][column] = "x";
+                        changes = true;
+                    }
+                }
+            }
+        } while (changes);
+
+        // check for winner
+        String winner = "";
+        // O wins at the bottom
+        for (int i = 0; i < columnCount; i++) {
+            if (values[lineCount - 1][i].equals("o")) winner = "O";
+        }
+        // X wins on the right
+        for (int i = 0; i < lineCount; i++) {
+            if (values[i][columnCount - 1].equals("x")) winner = "X";
+        }
+
+        return winner;
+    }
+
+    boolean checkSurroundingsFor(String[][] values, int line, int column, String character) {
```

Comment:
This method appears to be quite complicated and hard to read. Could you please rethink the implementation here?
For example it helps to extract more methods as I already asked to do in the code above.

In the C# track is a good example of an implementation that uses many small methods and more descriptive method and variable names to improve readability.